hi-ml/src/health_ml/utils/regression_test_utils.py

```diff
@@ -71,6 +114,13 @@ def try_read_csv(prefix: str, file: Path) -> Optional[pd.DataFrame]:
         if expected_lines != actual_lines:
             print_lines("Expected", expected_lines)
             print_lines("Actual", actual_lines)
+            # Add additional context for json file mismatches
+            if expected.suffix == ".json":
+                compare_dictionaries(
+                    _load_json_from_text_lines(expected_lines),
+                    _load_json_from_text_lines(actual_lines),
+                    tolerance=csv_relative_tolerance
+                )
```

Comment:
Completely different idea, should we print out a generic diff here, like https://stackoverflow.com/questions/44348040/python-diff-two-multiline-strings-like-github, using difflib?